cleanlab/multiannotator.py

```diff
@@ -0,0 +1,436 @@
+) -> np.array:
+    """Returns the fractions of annotators that agree with the consensus labels. Note that the
+    fraction for each example only considers the annotators that labeled that particular example.
+
+    Parameters
+    ----------
+    labels_multiannotator : pd.DataFrame
+        2D pandas DataFrame of multiple given labels for each example with shape (N, M),
+        where N is the number of examples and M is the number of annotators.
+        For more details, labels in the same format expected by the :py:func:`get_label_quality_scores_multiannotator <cleanlab.multiannotator.get_label_quality_scores_multiannotator>`.
+
+    consensus_labels : np.array
+        An array of shape ``(N,)`` with the consensus labels aggregated from all annotators.
+
+    Returns
+    -------
+    annotator_agreement : np.array
+        An array of shape ``(N,)`` with the fraction of annotators that agree with each consensus label.
+    """
+    return (
+        labels_multiannotator.assign(consensus_label=consensus_labels)
+        .apply(
+            lambda s: np.mean(s.drop("consensus_label") == s["consensus_label"]),
+            axis=1,
+        )
+        .to_numpy()
+    )
+
+
+# TODO: add other ways to the get quality score (methods that can be passed to get_label_quality_score)
+def get_quality_of_consensus(
+    consensus_labels: np.array,
+    pred_probs: np.array,
+    num_annotations: np.array,
+    annotator_agreement: np.array,
+    kwargs: dict = {},
+) -> np.array:
+    """Return scores representing quality of the consensus label for each example,
+    calculated using weighted product of `label_quality_score` of `consensus_labels` and `annotator_agreement`
+
+    Parameters
+    ----------
+    consensus_labels : np.array
+        An array of shape ``(N,)`` with the consensus labels aggregated from all annotators.
+
+    pred_probs : np.array
+        An array of shape ``(N, K)`` of model-predicted probabilities, ``P(label=k|x)``.
+        For details, predicted probabilities in the same format expected by the :py:func:`get_label_quality_scores_multiannotator <cleanlab.multiannotator.get_label_quality_scores_multiannotator>`.
+
+    num_annotations : np.array
+        An array of shape ``(N,)`` with the number of annotators that have labeled each example.
+
+    annotator_agreement : np.array
+        An array of shape ``(N,)`` with the fraction of annotators that agree with each consensus label.
+
+    Returns
+    -------
+    quality_of_consensus : np.array
+        An array of shape ``(N,)`` with the quality score of the consensus.
+        TODO: better explanation of this
+    """
+    lqs_consensus_labels = get_label_quality_scores(consensus_labels, pred_probs, **kwargs)
+    frac = pred_probs[range(len(consensus_labels)), (consensus_labels)] / num_annotations
+    quality_of_consensus = frac * lqs_consensus_labels + (1 - frac) * annotator_agreement
+    return quality_of_consensus
+
+
+def compute_consensus_stats(
```

Comment:
Let's make this helper method (with underscore prefix): `_compute_consensus_stats()`

So we don't have to commit to supporting this exact API forever.